presto-main/src/test/java/io/prestosql/server/ui/TestWebUi.java

```diff
@@ -344,13 +344,115 @@ private static void assertRedirect(OkHttpClient client, String url, String redir
         if (testProxy) {
             request = new Request.Builder()
                     .url(url)
-                    .header(X_FORWARDED_PROTO, "https")
+                    .header(X_FORWARDED_PROTO, "test")
                     .header(X_FORWARDED_HOST, "my-load-balancer.local")
-                    .header(X_FORWARDED_PORT, "443")
+                    .header(X_FORWARDED_PORT, "123")
                     .build();
             try (Response response = client.newCall(request).execute()) {
                 assertEquals(response.code(), SC_SEE_OTHER);
-                assertEquals(response.header(LOCATION), "https://my-load-balancer.local:443/" + redirectLocation.replaceFirst("^([^/]*/){3}", ""));
```

Comment:
i deliberately used .replace instead of using uri builder. 
Replace is simplistic, so it's obvious what the expected value is.
With uri builder used in the test and in the code under test, it's not as obvious.

Reply:
I am confused by this statement.  The regex to replace sub parts in the middle of the string would be very complex.  I find the new code really clear:
```
uriBuilderFrom(URI.create(redirectLocation))
                    .scheme("test")
                    .host("my-load-balancer.local")
                    .port(123)
                    .toString());
```
It literally says, take the redirect location, and replace some bits.